File: backend/service/src/main/java/org/example/service/auth/facade/AuthServiceImpl.java
package org.example.service.auth.facade;

import org.example.model.projection.login.LoginResponse;
import org.example.model.request.login.FacebookLoginRequest;
import org.example.model.request.login.LoginRequest;
import org.example.service.auth.login.facebook.FacebookLoginService;
import org.example.service.auth.login.manual.ManualLoginService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class AuthServiceImpl implements AuthService {
    private final ManualLoginService manualLoginService;
    private final FacebookLoginService facebookLoginService;

    @Autowired
    public AuthServiceImpl(ManualLoginService manualLoginService, FacebookLoginService facebookLoginService) {
        this.manualLoginService = manualLoginService;
        this.facebookLoginService = facebookLoginService;
    }

    @Override
    public LoginResponse login(LoginRequest request) {
        return manualLoginService.execute(request);
    }

    @Override
    public LoginResponse loginByFacebook(FacebookLoginRequest request) {
        return facebookLoginService.execute(request);
    }
}

File: laptop-store-backend/service/src/main/kotlin/org/example/service/auth/AuthServiceImpl.kt
package org.example.service.auth

import org.example.exception.InvalidCredentialException
import org.example.model.input.LoginInput
import org.example.model.response.LoginResponse
import org.example.security.JwtProvider
import org.example.validator.auth.AuthValidator
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class AuthServiceImpl @Autowired constructor(
    private val authValidator: AuthValidator,
    private val jwtProvider: JwtProvider
) : AuthService {
    override fun login(loginInput: LoginInput): LoginResponse {
        if (isInvalidLoginInput(loginInput)) {
            throw InvalidCredentialException()
        }
        val username = loginInput.username!!
        val (accessToken, refreshToken) = jwtProvider.getAccessAndRefreshTokens(username)
        return LoginResponse(accessToken = accessToken, refreshToken = refreshToken)
    }

    private fun isInvalidLoginInput(loginInput: LoginInput): Boolean {
        return !authValidator.validateLoginInput(loginInput)
    }
}